.circleci/config.yml

```diff
@@ -0,0 +1,142 @@
+  docker:
+  - image: ucbbar/chisel3-tools
+    user: "chisel"
+
+  working_directory: ~/repo
+
+  environment:
+    # Customize the JVM maximum heap limit
+    JVM_OPTS: -Xmx3200m
+    TERM: dumb
+    CHISEL_REV: origin/master
+    FIRRTL_REPO: git@github.com:freechipsproject/firrtl.git
+    FIRRTL_BRANCH: master
+    FIRRTL_REV: master
+    SBT_ARGS: ""
+
+
+version: 2
+
+jobs:
+  build-prep:
+    <<: *defaultImageUserWDEnv
+
+    steps:
+
+    # Perform a "default" checkout to get ssh set up.
+    - checkout:
+        path: chisel3
+
+    - run:
+        command: |
+          date > date.prep
+          printenv
+          ls -l
+          git clone --depth 10 --branch $FIRRTL_BRANCH "$FIRRTL_REPO" firrtl && (cd firrtl && git checkout $FIRRTL_REV)
+          echo $FIRRTL_REV && (cd firrtl && git log -1 > ../firrtl.log)
+
+    - save_cache:
+        paths:
+        - ~/.ssh
+        key: v1-dependencies-{{ .Branch }}-{{ .Revision }}-{{ checksum "firrtl.log" }}
+
+    - persist_to_workspace:
+        root: /home/chisel
+        paths:
+        - repo
+        - .ivy2
+        - .m2
+        - .sbt
+
+  build-firrtl:
+    <<: *defaultImageUserWDEnv
+
+    steps:
+    - attach_workspace:
+        at: /home/chisel
+
+    - restore_cache:
+        keys:
+        - v1-dependencies-{{ .Branch }}-{{ .Revision }}-{{ checksum "firrtl.log" }}
+        # fallback to using the latest cache if no exact match is found
+        - v1-dependencies-
+
+    # publish FIRRTL
+    - run:
+        command: |
+          date > date.firrtl
+          (cd firrtl && cat /dev/null | sbt $SBT_ARGS +publishLocal)
+
+    - persist_to_workspace:
+        root: /home/chisel
+        paths:
+        - repo
+        - .ivy2
+        - .m2
+        - .sbt
+
+  build-chisel:
+    <<: *defaultImageUserWDEnv
+
+    steps:
+    - attach_workspace:
+        at: /home/chisel
+    # Download and cache dependencies
+    - restore_cache:
+        keys:
+        - v1-dependencies-{{ .Branch }}-{{ .Revision }}-{{ checksum "firrtl.log" }}
+        # fallback to using the latest cache if no exact match is found
+        - v1-dependencies-
+
+    # publish chisel
+    - run: date > date.chisel3 && (cd chisel3 && sbt $SBT_ARGS +publishLocal)
+
+    - persist_to_workspace:
+        root: /home/chisel
+        paths:
+        - repo
+        - .ivy2
+        - .m2
+        - .sbt
+
+  test-chisel:
+    <<: *defaultImageUserWDEnv
+
+    steps:
+    - attach_workspace:
+        at: /home/chisel
+    - restore_cache:
+        keys:
+        - v1-dependencies-{{ .Branch }}-{{ .Revision }}-{{ checksum "firrtl.log" }}
+        # fallback to using the latest cache if no exact match is found
+        - v1-dependencies-
+
+    # Set environment
+    - run: echo 'export PATH="/opt/verilator/verilator_3_922/bin:/opt/yosys/bin:$PATH"' >> $BASH_ENV
+    # We seem to have memory leaks in chisel tests -
+    # we run out of memory trying to "sbt +test" but a single "sbt test" passes.
```

Comment:
Anecdotally, I only started seeing this locally with Scala 2.12. Good to include this in the documentation here.